driver/terraform.go

```diff
@@ -488,6 +501,9 @@ func (tf *Terraform) inspectTask(ctx context.Context, returnPlan bool) (InspectP
 			fmt.Sprintf("error tf-plan for '%s'", taskName))
 	}
 
+	// Delete the task once the inspection has completed
+	tf.deleteTask(ctx)
+
```

Comment:
I was looking to understand where the template is registered. It looks like it's in `initTaskTemplate()` which is called by `initTask()`. `UpdateTask()` calls `initTask` and `inspectTask`. What are your thoughts on moving this template deregistration call to `UpdateTask()`? I think it would add a little parity to the watcher registration and deregistration. But not sure if there's nuance I'm missing

Reply:
Just to clarify, you want me to look into including the call in UpdateTask() as well?

Reply:
I wanted to get your input on the possibility of moving `tf.deleteTask(ctx)` out of `inspectTask()` and into `UpdateTask()` so that `deleteTask()` is called below `inspectTask()`. See what your thoughts are and if it's even possible

Reply:
Still not sure I quite understand, but InspectTask() which calls inspectTask() is called by controller/readwrite.go createTask, which doesn't call UpdateTask(). So the tf.deleteTask(ctx) needs to be part of inspectTask() method.

Reply:
deleteTask needs to be called wherever an inspect is done

Reply:
For posterity, we chatted about this async (feel free to correct if I misrepresent anything):
- Talked about renaming the private method to `deregisterTemplate()`
- There are instances where inspectTask() may not want to deregister the template, so deregisterTemplate can be moved outside of inspectTask